db/system_keyspace.cc

```diff
@@ -1569,12 +1569,26 @@ data_value system_keyspace::make_tokens_data_value(const std::unordered_set<dht:
     return make_set_value(set_type, prepare_tokens(tokens));
 }
 
+static locator::host_id get_host_id(const gms::inet_address& peer, const cql3::untyped_result_set::row& row) {
+    locator::host_id host_id;
+    if (row.has("host_id")) {
+        host_id = locator::host_id(row.get_as<utils::UUID>("host_id"));
+    }
+    if (!host_id) {
+        slogger.warn("Peer {} has no host_id in system.{}", peer, system_keyspace::PEERS);
+    }
+    return host_id;
+}
+
 future<std::unordered_map<gms::inet_address, std::unordered_set<dht::token>>> system_keyspace::load_tokens() {
-    sstring req = format("SELECT peer, tokens FROM system.{}", PEERS);
+    sstring req = format("SELECT peer, host_id, tokens FROM system.{}", PEERS);
```

Comment:
How does this handle mixed clusters?

Reply:
All supported versions already store host_id in system.peers.  It shouldn't be a problem.

Reply:
When did we start writing it?

Reply:
4311662828aa09627c77d9af16478f22624b6a25 called `update_peer_info` from `storage_service::handle_state_normal` back in `0.10`.
Note the use of `gossiper::uses_host_id` which was true from inception (See 6e83e54b0d61adc596a90e30d7ec4674d2176863 where I got rid of it)

The pretty-much initial implementation of `prepare_to_join` at efe067fd749626480c45ddf34de62a8853aa162c already populated the `HOST_ID` in the endpoint state, and it is sent in gossip also basically from inception (eed2795abe99ffc1619f7170707032e34777e350).

e86d39faf021f2797fdbbca93dfa9016e9c72430 moved the call to `update_peer_info` a bit later in `handle_state_normal`, when the node becomes a normal token owner (in branch 4.6)
but it doesn't matter since we've wrote the host_id to system.peers even before that.

Reply:
ok, so no compatibility issues.